Xamarin.Forms.Core/DependencyResolver.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Linq;
+using System.Reflection;
+
+namespace Xamarin.Forms.Internals
+{
+	public static class DependencyResolver
+	{
+		public delegate object ResolveDelegate(Type type, params object[] args);
```

Comment:
Suggest we use `Func<Type, object[], object>`; Suggest we conserve the number of new Types introduced at the expense syntactic sugar at our call site; We can just put the args into a new object[] and save ourselves a bit of public surface area, no?

Reply:
Done.